state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -2027,8 +2043,13 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 
 		let channel_owner = deposit_channel_details.owner.clone();
 
-		if let Some(tx_id) = deposit_details.deposit_id() {
-			if TransactionsMarkedForRejection::<T, I>::take(&channel_owner, &tx_id).is_some() &&
+		if let Some(tx_ids) = deposit_details.deposit_ids() {
+			let tainted_tx_id = Self::check_if_deposit_is_tainted(
```

Comment:
No need for a function / iterating the prefix. 

We can do it a bit like before:

```rust
if tx_ids
	.iter()
	.filter_map(|tx_id|
		TransactionsMarkedForRejection::<T, I>::take(
			&channel_owner,
			tx_id
		)
	)
	.next()
	.is_some() &&
	// ..
```

This way we will also remove *all* tx_ids that match and we don't need the call to `remove` below.